### DISCUSSION TITLE
Question about editing Existing Images

### DISCUSSION THREAD

**Discussion by Scott-lasertrade** [rpldy/react-uploady#404] [General]
Hi,

I'm not sure if I'm just not finding the right documentation or it's out of scope for react-uploady's feature set. 

**Context** 
I've mostly followed/set up the Second Part example from the guide here - https://www.youtube.com/watch?v=a6FHlPhw6Vg 
(React-cropper with multiple images)

The main difference beyond css/visual changes is I'm utilizing AWS  s3 for my destination, and I override the useRequestPreSend to write to different specified urls per photo. This seems to work fine. 

Ideally I'd also want to allow editing of existing images, can that be part of react-uploady? I.e. Can I pull the image files/urls down from aws s3 and plug that into the initial data somehow? So that I can further edit/change these images, on top of uploading new images, or should this process be a completely different component? 

**TL:DR** 
Is there a way to plug existing/intial images into react-uploady so that images can be edited.

**Comment by yoavniran** [comment]
Hey @Scott-lasertrade 

What you're describing is completely outside Uploady's scope. Uploady focuses on uploading data (usually files) to an endpoint. 

Downloading and editing (or any other non-upload related) tasks is intentionally left out as there are many fine tools out there and to keep this library small and focused.

Once you have the image you edited in the client it should be very easy to re-upload it back to AWS if I understand you correctly. You can supply the blob as the upload data.

**Comment by Scott-lasertrade** [comment]
Thank you for the swift reply @yoavniran, good to know I haven't missed anything obvious.

With the method you have mentioned above, let's say I have already have the separate object that manages pulling down the image, editing it and producing a final blob, how would I then go about programatically supplying that blob data to Uploady?

**Reply by yoavniran** [reply]
[Uploady's context upload](https://react-uploady.org/docs/api/context/#upload) method is the way to go.
it will accept the blob

**Reply by Scott-lasertrade** [reply]
Awesome, wll give it a go, thanks for your help!
